src/libstd/sync/mpsc/mod.rs

```diff
@@ -544,36 +551,46 @@ impl<T> UnsafeFlavor<T> for Receiver<T> {
 }
 
 /// Creates a new asynchronous channel, returning the sender/receiver halves.
-/// All data sent on the sender will become available on the receiver, and no
-/// send will block the calling thread (this channel has an "infinite buffer").
+/// All data sent on the [`Sender`] will become available on the [`Receiver`] in
```

Comment:
Expanding the docs on channel - most of this is adapted from the mspc module page since that's where the bulk of this is explained.